packages/ember-application/tests/system/visit_test.js

```diff
@@ -35,6 +37,67 @@ moduleFor('Application - visit()', class extends ApplicationTestCase {
     );
   }
 
+  [`@test does not add serialize-mode markers by default`](assert) {
+    let templateContent = '<div class="foo">Hi, Mom!</div>';
+    this.addTemplate('index', templateContent);
+    let rootElement = document.createElement('div');
+
+    let bootOptions = {
+      isBrowser: false,
+      rootElement
+    };
+
+    ENV._APPLICATION_TEMPLATE_WRAPPER = false;
+    return this.visit('/', bootOptions).then(()=> {
+      assert.equal(rootElement.innerHTML, templateContent, 'without serialize flag renders as expected');
+    });
+  }
+
+  [`@test _renderMode: rehydration`](assert) {
+    assert.expect(2);
+
+    let indexTemplate = '<div class="foo">Hi, Mom!</div>';
+    this.addTemplate('index', indexTemplate);
+    let rootElement = document.createElement('div');
+
+    let bootOptions = {
+      isBrowser: false,
+      rootElement,
+      _renderMode: 'serialize'
+    };
+
+    ENV._APPLICATION_TEMPLATE_WRAPPER = false;
+
+    return this.runTask(() => {
```

Comment:
we should remove this runTask (otherwise this makes a nested run loop and we shouldn't need that)